extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/CassandraCluster.java

```diff
@@ -0,0 +1,89 @@
+package org.dataportabilityproject.cloud.microsoft.cosmos;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.RemoteEndpointAwareJdkSSLOptions;
+import com.datastax.driver.core.SSLOptions;
+import com.datastax.driver.core.Session;
+
+import javax.net.ssl.SSLContext;
+
+/**
+ * Configures a client connection to a Cassandra cluster.
+ */
+public class CassandraCluster {
+    private Cluster cluster;
+    private String host = "127.0.0.1";
```

Comment:
Yeah, this needs to go as soon as I understand how configuration is obtained. Is there an API that provides access to the YAML properties as opposed to getResources() from the classpath?

This may be part of the further modularity discussions. For example,  a way for extensions to obtain references to SPI services and keyed properties during bootstrap? Also, are there already conventions around namespacing configuration keys so extensions don't collide?

Reply:
Yeah, currently we copy the yaml settings into our jar at $SRC_DIR/src/main/resources (see https://github.com/google/data-portability/blob/master/bin/build_and_run_jar.sh#L40) and expose to java code via CommonSettings (https://github.com/google/data-portability/blob/c9f95e82b44eaee627562ddb543462272fef6f3a/portability-core/src/main/java/org/dataportabilityproject/PortabilityCoreModule.java#L63)

Does that model make sense with the new design?

Reply:
Forgot to mention we are somewhat modular in how we do this now by selectively including different yaml files based on which binary. e.g. for the API, we include api.yaml as well as common.yaml. For the worker, we include only common.yaml (we don't have any worker-only settings as of yet).

If we want to keep using the build script in the new design, we can continue to selectively load in settings based on service provider, cloud provider and so on.

Reply:
This is probably part of the ongoing modularity discussion. 

I think we should be able to handle runtime assembly (getting the bits into a file system layout and then possibly into a container image) from Gradle.  

My preference would be to have a bootstrap API for each subsystem that takes CommonSettings as a parameter. Runtime launchers could then decide how to source configuration, e.g. YAML, environment properties, build script params, a combination, etc. There would be a "provided" launcher that used YAML but custom ones could be written, That would give users flexibility in how they want to mange deployment and configuration.

Reply:
Feel free to add this as an issue